tests/unit/testsuite/Bolt/Boltpay/Model/OrderTest.php

```diff
@@ -2391,4 +2391,23 @@ public function activateOrder_withParentQuoteRemovalProblems_logsAndNotifiesExce
             )
         );
     }
+
+    /**
+     * @test
+     * @throws ReflectionException
+     */
+    public function applyShipmentToQuoteFromTransaction_withTransactionIsMissingShipmentsProperty()
+    {
+        $result =  TestHelper::callNonPublicFunction(
+            $this->currentMock,
+            'applyShipmentToQuoteFromTransaction',
+            array(
+                $this->immutableQuoteMock,
+                new stdClass(),
+                1
+            )
+        );
+
+        $this->assertNull($result);
+    }
```

Comment:
For me, this is very close to a placeholder test.
Perhaps, it is better named something like `applyShipmentToQuoteFromTransaction_withVirtualCart_doesNotApplyShipment` and certainly would be better to mock the `Bolt_Boltpay_Model_ShippingAndTax` class and stub `applyBoltAddressData`.  Then you can `expect` that the method is never called.

The `$result` and `$this->assertNull($result)` does not make any real semantic sense and I assume added in order to circumvent the PHPUnit "meaningless test" error that would otherwise be thrown for setting no expectations or assertions.

To stub the model, use the static method `TestHelper::stubModel`.  Make sure that either `TestHelper::restoreModel` or `TestHelper::restoreOriginals` is subsequently called in either the same method or a tearDown.